texar/torch/data/tokenizers/t5_tokenizer.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Pre-trained T5 tokenizer.
+
+"""
+
+from typing import Any, Dict, Optional
+
+import os
+import re
+
+from texar.torch.data.tokenizers.sentencepiece_tokenizer \
+    import SentencePieceTokenizer
+from texar.torch.modules.pretrained.t5 import PretrainedT5Mixin
+
+__all__ = [
+    'T5Tokenizer',
+]
+
+
+class T5Tokenizer(SentencePieceTokenizer, PretrainedT5Mixin):
+    r"""Pre-trained T5 Tokenizer.
+
+    Args:
+        pretrained_model_name (optional): a `str`, the name of
+            pre-trained model (e.g., `T5-Small`). Please refer to
+            :class:`~texar.torch.modules.PretrainedT5Mixin` for
+            all supported models.
+            If None, the model name in :attr:`hparams` is used.
+        cache_dir (optional): the path to a folder in which the
+            pre-trained models will be cached. If `None` (default),
+            a default directory (``texar_data`` folder under user's home
+            directory) will be used.
+        hparams (dict or HParams, optional): Hyperparameters. Missing
+            hyperparameters will be set to default values. See
+            :meth:`default_hparams` for the hyperparameter structure
+            and default values.
+    """
+
+    _IS_PRETRAINED = True
+
+    _VOCAB_FILE_NAMES = {
+        'vocab_file': 'sentencepiece.model'
+    }
+
+    _MAX_INPUT_SIZE = {
+        'T5-Small': 512,
+        'T5-Base': 512,
+        'T5-Large': 512,
+        'T5-3B': 512,
+        'T5-11B': 512
+    }
+
+    def __init__(self,
+                 pretrained_model_name: Optional[str] = None,
+                 cache_dir: Optional[str] = None,
+                 hparams=None):
+
+        self.load_pretrained_config(pretrained_model_name, cache_dir, hparams)
+
+        if self.pretrained_model_dir is not None:
+            assert self.pretrained_model_name is not None
+            vocab_file = os.path.join(self.pretrained_model_dir,
+                                      self._VOCAB_FILE_NAMES['vocab_file'])
+            assert self.pretrained_model_name is not None
```

Comment:
why do we check it twice?

Reply:
We should'nt. Removing it from `BERTTokenizer` too.